pennylane/qchem/tapering.py

```diff
@@ -538,3 +539,142 @@ def taper_hf(generators, paulixops, paulix_sector, num_electrons, num_wires):
         tapered_hartree_fock.append(0)
 
     return np.array(tapered_hartree_fock).astype(int)
+
+
+# pylint: disable=too-many-branches, too-many-arguments, inconsistent-return-statements, no-member
+def taper_operation(operation, generators, paulixops, paulix_sector, wire_order, gen_op=None):
+    r"""Transforms a gate operation with a Clifford operator and taper qubits.
+
+    The qubit operator for the generator of the gate operation is computed either internally or can be provided
+    manually via `gen_op` argument. If this operator commutes with all the :math:`\mathbb{Z}_2` symmetries of
+    the molecular Hamiltonian, then this operator is transformed using the Clifford operators :math:`U` and
+    tapered, otherwise it is discarded. Finally, the tapered generator is exponentiated using :func:`~.PauliRot`
+    for building the tapered unitary.
+
+    Args:
+        operation (Operation): qubit operation to be tapered
+        generators (list[Hamiltonian]): generators expressed as PennyLane Hamiltonians
+        paulixops (list[Operation]):  list of single-qubit Pauli-X operators
+        paulix_sector (list[int]): eigenvalues of the Pauli-X operators
+        wire_order (Sequence[Any]): order of the wires in the quantum circuit
+        gen_op (Hamiltonian): optional argument to provide the generator of the operation
+
+    Returns:
+        list(Operation): list of operations of type :func:`~.PauliRot` implementing tapered unitary operation
+
+    Raises:
+        NotImplementedError: generator of the operation cannot be constructed internally
+        ValueError: operation needs to be a :class:`~.pennylane.Hamiltonian` and a valid generator of the operation
+
+    **Example**
+
+    >>> symbols, geometry = ['He', 'H'], np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.4589]])
+    >>> mol = qchem.Molecule(symbols, geometry, charge=1)
+    >>> H, n_qubits = qchem.molecular_hamiltonian(symbols, geometry)
+    >>> generators = qchem.symmetry_generators(H)
+    >>> paulixops = qchem.paulix_ops(generators, n_qubits)
+    >>> paulix_sector = qchem.optimal_sector(H, genera  tors, mol.n_electrons)
+    >>> qchem.taper_operation(qml.SingleExcitation(1, wires=[0, 2]),
```

Comment:
This Comment applies also to the example code in the changelog: why is the gate parameter `1`? I find that a confusing choice since I'd expect a float

Reply:
This was rather arbitrary. I am now updating it to `np.pi`.